spikeinterface/core/waveform_extractor.py

```diff
@@ -153,7 +157,56 @@ def register_extension(cls, extension_class):
         assert all(extension_class.extension_name != ext.extension_name for ext in cls.extensions), \
             'Extension name already exists'
         cls.extensions.append(extension_class)
+    
+    # map some method from recording and sorting
+    @property
+    def recording(self):
+        if self._recording is None:
+            raise(ValueError('WaveformExtractor is used in mode "without_recording=True" this operation need the recording'))
+        return self._recording
+
+    @property
+    def channel_ids(self):
+        if self._recording is not None:
+            return self.recording.channel_ids
+        else:
+            raise NotImplementedError
+    
+    @property
+    def sampling_frequency(self):
+        if self._recording is not None:
+            return self.recording.get_sampling_frequency()
+        else:
+            raise NotImplementedError
+    
+    def get_num_channels(self):
+        if self._recording is not None:
+            return self.recording.get_num_channels()
+        else:
+            raise NotImplementedError
+
+    def get_num_segments(self):
+        return self.sorting.get_num_segments()
```

Comment:
These properties should be saved, so that they can also be accessed if recording is None

Reply:
Plus we should have `get_channel_locations`

Reply:
Yep. Done